### PR TITLE
Make sure all data is returned as a list from explanation

### PR DESCRIPTION
**Pull request by wamartin-aml**

### CODE REVIEW

File: python/interpret_community/explanation/explanation.py [hunk 1]
```diff
@@ -1148,25 +1148,41 @@ def eval_data(self):
...
-        return self._eval_y_predicted_proba
+        return self._convert_to_list(self._eval_y_predicted_proba)
+
+    def _convert_to_list(self, data):
+        """Convert data to a Python list.
+
+        :param data: The data to be converted.
+        :type data: np.array, pd.DataFrame, list, scipy.sparse
+        :return: The data converted to a list (except for sparse which is unchanged).
+        :rtype: list | scipy.sparse
```
Comment (imatiach-msft) [inline python/interpret_community/explanation/explanation.py:1177]: for multiclass case:
| list[scipy.sparse]

File: python/interpret_community/explanation/explanation.py [hunk 2]
```diff
@@ -1148,25 +1148,41 @@ def eval_data(self):
...
-        return self._eval_data
+        return self._convert_to_list(self._eval_data)
 
     @property
     def eval_y_predicted(self):
         """Get predicted ys for the evaluation data.
 
         :return: The predicted ys for the evaluation data.
-        :rtype: np.array
+        :rtype: list
```
Comment (imatiach-msft) [inline python/interpret_community/explanation/explanation.py:1158]: do we need to specify type here like list[int | float] or not?

File: python/interpret_community/explanation/explanation.py [hunk 3]
```diff
@@ -1148,25 +1148,41 @@ def eval_data(self):
...
+        return self._convert_to_list(self._eval_y_predicted)
 
     @property
     def eval_y_predicted_proba(self):
         """Get predicted probability ys for the evaluation data.
 
-        :param eval_ys_predicted_proba: The predicted probability ys for the evaluation data.
-        :type eval_ys_predicted_proba: np.array
+        :return: The predicted probability ys for the evaluation data.
+        :rtype: list
```
Comment (imatiach-msft) [inline python/interpret_community/explanation/explanation.py:1167]: this is a list[list[float]], is that right?

**Comment by imatiach-msft** [comment]
did you have a chance to try this out with the explanation client locally to make sure it can still be uploaded/downloaded/visualized?

File: python/interpret_community/explanation/explanation.py (hunk 1 above)
Comment (wamartin-aml) [reply python/interpret_community/explanation/explanation.py:1177]: will update

File: python/interpret_community/explanation/explanation.py (hunk 2 above)
Comment (wamartin-aml) [reply python/interpret_community/explanation/explanation.py:1158]: will add a generic

File: python/interpret_community/explanation/explanation.py (hunk 3 above)
Comment (wamartin-aml) [reply python/interpret_community/explanation/explanation.py:1167]: added

**Comment by wamartin-aml** [comment]
I've confirmed the explanation client tests pass with these changes, checking in now

**Pull request merged**